18414 change usages of standard file stream and multi byte file stream to file  decorators

guillep opened the pull request:
Make the system use the new files:
 - remove usages of MultibyteFileStream, StandardFileStream and FileStream
 - pushing usage of Zn*Streams

On the way, I've fixed 64 bit tests... Decoupling them in two different issues is complicated...

guillep committed:
Adding read write buffered streams

guillep committed:
Enhancements on SourceArray, RemoteString and friends

- Adding tests for SourceArray, RemoteString and friends using an in memory file system
- a RemoteString should know the source array he uses instead of using a global
- extend Zn streams to handle position and setToEnd

guillep committed:
Files use buffered streams by default

guillep committed:
Making streams and Fuel tests pass

guillep committed:
Fix tests for 64 bits.

guillep committed:
Recategorization

guillep committed:
Fixing tests for 64 bits

guillep committed:
Fix stream tests

guillep committed:
Changes file write stream uses new streams

guillep committed:
Recovering temp names

guillep committed:
Using new streams for changes read stream also

guillep committed:
Refactor to set potential locations to file

guillep committed:
Using new source mechanism for sources file also

guillep committed:
Removed deprecated method

guillep committed:
Making it work for comments also

guillep committed:
Fix bug due to MultiByteFileStream bug

guillep committed:
Using New files

guillep committed:
Extending file system with binary streams

guillep committed:
- File copy should be a binary operation
- Make Ombu work with new streams
- extended stream API

guillep committed:
Making tests pass

guillep committed:
Add makefile to ease debugging

guillep committed:
Bootstrap with new files

guillep committed:
Last touches in bootstrap scripts

guillep committed:
Declaring temps

guillep committed:
Making tests >= R run green

guillep committed:
Making all tests except monticello run

guillep committed:
Monticello tests are green

guillep committed:
Fix zip extraction

guillep committed:
Add debug Makefile

guillep commented:
Yes I'm looking into it right now

guillep committed:
Jenkins file should specify BOOTSTRAP_REPOSITORY

guillep committed:
Cannot use bash syntax in jenkinsfile

guillep committed:
Set correct directory

guillep committed:
export BOOTSTRAP_REPOSITORY variable

guillep committed:
Fixing read write streams contents position

dionisiydk commented on src/Files-Prompt/MorphicUIManager.extension.st:
```diff
@@ -9,6 +9,7 @@ MorphicUIManager >> fileDoesNotExistUserHandling: filename [
 			('choose another name' translated).
 			('cancel' translated)}
 		title: filename, ' does not exist.'.
+	self halt.
```
The halt in code

dionisiydk reviewed (commented):
A lot of good changes. It can break something but it is important improvement

guillep committed:
- File streams should not be buffered by default
- File streams provided by file system should be buffered by default

guillep committed:
Updated makefile for latest build

guillep committed:
Fix tests

guillep committed:
contents should behave as in a write stream

dionisiydk commented:
In another issue https://pharo.fogbugz.com/f/cases/19026/Merge-FileDoesNotExist-and-FileDoesNotExistException we should clean file exception hierarchies. We have two now

guillep committed:
Fixing zip tests

guillep committed:
Fix Tonel tests

guillep committed:
Make sure write streams are closed

guillep committed:
FileSystem should delegate deletion to File

guillep committed:
ReadStream should be read only

guillep committed:
Remove debugging code

guillep committed:
Fix zip test in windows
 - close zip stream before deleting file.

guillep committed:
All zn streams should be flushable for users compatibility.

guillep committed:
Fix windows tests
 - file tests should close files
 - PNG tests should close files
 - SmalltalkImage and SourceFileArray should not open the source files.

guillep committed:
add InitializePackages command line handler into debugging makefile

guillep committed:
Do not close a Source file if it is not open
